### PR TITLE
Added "copy profile url" button

### LINKED ISSUE #569: Profile Url
**Is your feature request related to a problem? Please describe.**
When I want to invite some friends to check it out.  I'd like to be able to link to a profile link.  like: https://dogehouse.tv/user/username

This way they could follow me and I could invite them to a private room.  Otherwise I have to make it public get them to join.  Follow them.. And then create a private room and bring them all in.

**Describe the solution you'd like**
It could just show the little follow card over the channel list if need be.  Just a way that they could find me and then follow

**Describe alternatives you've considered**
Search would also work

**Additional context**

### PR DESCRIPTION
**Pull request by rajpatil53**
On clicking "copy profile url" button, profile URL (/user/:username) will be copied to clipboard.

On visiting this URL, profile page will be displayed and logged in user can follow/unfollow. If the URL is for logged in user then "edit profile" button will be displayed instead of follow/unfollow

Fixes benawad/dogehouse#569

### CODE REVIEW

File: kofta/src/app/components/UserProfile.tsx [hunk 1]
```diff
@@ -81,40 +82,51 @@ export const UserProfile: React.FC<UserProfileProps> = ({
...
-                  op: "follow",
-                  d: {
-                    userId: profile.id,
-                    value: !youAreFollowing,
-                  },
-                });
-                setYouAreFollowing(!youAreFollowing);
-                onFollowUpdater(setCurrentRoom, me, profile);
+                if(copyTextToClipboard(profileUrl)){
+                  toast("URL copied to clipboard", { type: "success" });
```
Comment (wes337) [inline kofta/src/app/components/UserProfile.tsx]: Should toast messages also be translated?

**Review by HarrisonMayotte — APPROVED** [review]
Looks good to me.
Edit: Missed the above review- that should probably be a translation string as well.

File: kofta/src/app/pages/ViewUserPage.tsx [hunk 2]
```diff
@@ -19,22 +20,29 @@ export const ViewUserPage = () => {
...
-    () => {
-      return wsFetch<any>({
-        op: "get_user_profile",
-        d: { userId: username },
-      });
-    },
-    { enabled: !profileFromState && status === "auth-good" && !!username }
+  const isDirectFromUrl = !profileFromState && !!username;
+
+  const { isLoading, data: profileFromDB } = useUserProfileQuery(username , isDirectFromUrl && status === "auth-good" );
```
Comment (wes337) [inline kofta/src/app/pages/ViewUserPage.tsx:25]: What is happening here? We are fetching the profile info if the user comes in directly/externally from a URL?

Comment (rajpatil53) [reply kofta/src/app/pages/ViewUserPage.tsx:25]: Yes
First fetch user profile by username, then fetch if current logged in user is following that user

File: kofta/src/app/components/UserProfile.tsx (hunk 1 above)
Comment (rajpatil53) [reply kofta/src/app/components/UserProfile.tsx]: Did the change

**Review by wes337 — APPROVED** [review]

**Review by HarrisonMayotte — APPROVED** [review]

**Comment by jamesql** [comment]
fixed merge conflicts with the Greek translation file

**Comment by jamesql** [comment]
@HarrisonMayotte are we good to merge this?

**Comment by HarrisonMayotte** [comment]
Sure.

**Comment by benawad** [comment]
@HarrisonMayotte in the future, wait to merge until I review it when there are backend changes

**Comment by HarrisonMayotte** [comment]
Sorry about that. Will do.

**Pull request merged**